generators/git/index.js

```diff
@@ -56,7 +62,10 @@ module.exports = class extends Generator {
     if (this.originUrl) {
       repository = this.originUrl;
     } else {
-      repository = this.options.githubAccount + '/' + this.options.name;
+      repository =
+        this.options.githubAccount +
+        '/' +
+        (this.options.repositoryName || this.options.name);
```

Comment:
Is there any case where `repositoryName` would be null and `name` have a value? I'd prefer we keep a single option here and let the parent generator pass the proper value.

Reply:
They could be falsy both if some other (non-node) generator would compose with this one or if the user would use it directly through `yo node:git`.

I’ve just tested this—both `githubAccount` and `repositoryName` are marked as required, yet there is no error when omitting them.

I agree with removing this fallback, as this is currently the only position at which `name` is used, and passing an option as a fallback for the case in which another option is not passed seems absurd.

There are, as I see it, three ways to deal with the possibility of having `undefined/undefined` as the repository:

* Only run `git init --quiet` when some options are missing
* Just ignore it
* Prompt for the missing values (which would only work when composing with this generator before `prompting`)

Reply:
I’ve implemented the first of the three options for now.